lib/Controller/PersonalSettingsController.php

```diff
@@ -0,0 +1,115 @@
+<?php
+declare(strict_types=1);
+
+namespace OCA\GPodderSync\Controller;
+
+use DateTime;
+
+use OCA\GPodderSync\Service\PodcastCacheService;
+use OCA\GPodderSync\Db\EpisodeAction\EpisodeActionRepository;
+use OCA\GPodderSync\Db\SubscriptionChange\SubscriptionChangeEntity;
+use OCA\GPodderSync\Db\SubscriptionChange\SubscriptionChangeRepository;
+
+use OCP\AppFramework\Controller;
+use OCP\AppFramework\Http\JSONResponse;
+use OCP\IL10N;
+use OCP\IRequest;
+use OCP\Settings\ISettings;
+
+use Psr\Log\LoggerInterface;
+
+class PersonalSettingsController extends Controller {
+
+	private LoggerInterface $logger;
+	private string $userId;
+	private SubscriptionChangeRepository $subscriptionChangeRepository;
+	private EpisodeActionRepository $episodeActionRepository;
+	private PodcastCacheService $podcastCacheService;
+
+	public function __construct(
+		string $AppName,
+		IRequest $request,
+		LoggerInterface $logger,
+		string $UserId,
+		SubscriptionChangeRepository $subscriptionChangeRepository,
+		EpisodeActionRepository $episodeActionRepository,
+		PodcastCacheService $podcastCacheService,
+	) {
+		parent::__construct($AppName, $request);
+		$this->logger = $logger;
+		$this->userId = $UserId ?? '';
+		$this->subscriptionChangeRepository = $subscriptionChangeRepository;
+		$this->episodeActionRepository = $episodeActionRepository;
+		$this->podcastCacheService = $podcastCacheService;
+	}
+
+	/**
+	 *
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
+	 *
+	 * @return JSONResponse
+	 */
+	public function metrics(): JSONResponse {
```

Comment:
Please extract all this code out of the controller to separate class(es). 
Then please add tests for those